src/libmongoc/tests/test-mongoc-client.c

```diff
@@ -2101,21 +2101,44 @@ _test_mongoc_client_ipv6 (bool pooled)
    mongoc_uri_t *uri;
    mongoc_client_pool_t *pool = NULL;
    mongoc_client_t *client;
+#if (defined(__APPLE__) || defined(_WIN32)) && defined(MONGOC_ENABLE_SSL)
+   mongoc_ssl_opt_t ssl_opts;
+#endif
    bson_error_t error;
 
    uri_str = test_framework_add_user_password_from_env ("mongodb://[::1]/");
    uri = mongoc_uri_new (uri_str);
    BSON_ASSERT (uri);
 
+#if (defined(__APPLE__) || defined(_WIN32)) && defined(MONGOC_ENABLE_SSL)
+   /* This is necessary because macOS & Windows seem to treat [::1] as not
+    * matching the hostname "0:0:0:0:0:0:0:1", which results in a certificate
+    * trust failure. */
```

Comment:
Can you create tickets to fix the hostname matching on IPv6 literals. Since the fixes for macOS and Windows would be separate, I think separate tickets makes sense. And note the ticket numbers in this comment so we can remove this once those are fixed.

Reply:
I wrote CDRIVER-3765 and CDRIVER-3766 and also updated the comment.  I'll push that change along with any others that are needed to finalize this PR.